app/models/concerns/tagable.rb

```diff
@@ -50,16 +50,17 @@ def update_tags(ids)
     self
   end
 
-  def tag(name_or_id)
+  def tag(name_or_id, user_id = APP_CONFIG['system_user_id'])
     Tagging.find_or_create_by(tag_id:         parse_tag_id!(name_or_id),
                               tagable_class:  self.class.name,
-                              tagable_id:     self.id)
+                              tagable_id:     id,
+                              last_user_id:   user_id)
     self
```

Comment:
that's better. will do.

Reply:
also better (for later) not storing user info on resource models so we don't have to reason about stuff like this! (when i save a tag, i only save tag-related stuff... something with a scope outside of a tag that composes tags with users takes care of updating user-related and version-related info)

Reply:
hmm... thinking about your proposed solution. in that scenario, doesn't it make `#tag` no longer idempotent?

to wit: if User A tags Goldman Sachs with `finace` and then an hour later User B tags Goldman Sachs finance, this logic will update the tag to be applied by User B. Which doesn't really make sense since they didn't really "update" the tag by changing its state in any meaningful way. the more sensible thing would be for nothing to happen...

what do you think?